src/dune/inline_tests.ml

```diff
@@ -280,7 +282,12 @@ include Sub_system.Register_end_point (struct
       Compilation_context.create () ~super_context:sctx ~expander ~scope
         ~obj_dir ~modules ~opaque:(Explicit false) ~requires_compile:runner_libs
         ~requires_link:(lazy runner_libs)
-        ~flags:(Ocaml_flags.of_list [ "-w"; "-24"; "-g" ])
+        ~flags:
+        (Ocaml_flags.of_list [ "-w"; "-24"; "-g" ])
+        ~compile_flags:
+        (Ocaml_flags.append_common
+           (Super_context.ocaml_flags sctx ~dir lib.buildable.flags)
```

Comment:
Looks like this is the problem. `lib.buildable.flags` corresponds to the flags attached to the `library` stanza. These are used to compiler the various modules of the library. Here, we want to use the flags specifically given by the user to compiler the test runner, which should be in `info.compile_flags`.

Reply:
which means `compile_flags` is of type `Ocaml_flags.Spec.t` and following this path, I again hit 
```
+  File "dune", line 11, characters 5-18:
+  11 |     (compile_flags -flag-that-is-not-accepted-by-ocaml)))
+            ^^^^^^^^^^^^^
+  Error: Unknown field compile_flags
```

Reply:
Indeed, is it clear how to make progress?

The idea is that when you write `let+ <var> = flag "<name>" decoder` or `and+ <var> = flag "<name>" decoder`, you are telling how to parse a field named `<name>`. `decoder` explains how to parse the argument of this field. `Ocaml_flags.Spec.decode` explains how to parse a set of three field and instead we want to parse a single `compile_flags` field of type `https://discuss.ocaml.org/t/ppx-omp-2-0-0-and-next-steps/6231`

Reply:
> Indeed, is it clear how to make progress?

now it seems clear, thanks.

Reply:
@jeremiedimino, am afraid, having `compile_flags` field to be of type `Ordered_set_lang.Unexpanded.t` will require us to pass `lib.buildable.flags` as a third arg to `Super_context.ocaml_flag` ( which expects the arg to be of type `Ocaml_flags.Spec.t` ), yet we intend to use `info.compile_flags` as you suggested above. This will call for changes i the type of `Super_context.ocaml_flag` which will subsequently result in `type` changes in many files.

Reply:
If you look at `Ocaml_flags.Spect.t`, it is a set of three things:

- a list of flags to pass to `ocamlc`
- a list of flags to pass to `ocamlopt`
- a common list of flags to pass to both `ocamlc` and `ocamlopt`

These three fields have the type `Ordered_set_lang.Unexpanded.t`. Since we are only adding one field `compile_flags`, it should correspond to a common list of flags to pass to both `ocamlc` and `ocamlopt`. So what we need is a way to create an `Ocaml_flags.Spec.t` value from a list of common flags. You can add a new function in `Ocaml_flags.Spec` for that.